src/client.cpp

```diff
@@ -269,260 +269,99 @@ void ADDON_Destroy()
   m_CurStatus = ADDON_STATUS_UNKNOWN;
 }
 
-ADDON_STATUS ADDON_SetSetting(const char *settingName, const void *settingValue)
+ADDON_STATUS SetStringSetting(const string settingName, const char* settingValue, string currentValue, ADDON_STATUS returnValueIfChanged)
```

Comment:
You should always keep the `std::` namespace. This improves readability and is common sense. `string` -> `std::string`.

And prefer by-reference parameters over by-value. `string`-> `const std::string&`

`currentValue`seems to be an inout param. Thus it must not be a by-value parameter!

Corrected signature: `ADDON_STATUS SetStringSetting(const std::string& settingName, const char* settingValue,  std::string& currentValue, ADDON_STATUS returnValueIfChanged)`